OSDOCS-9751: Added and described machine pool graphics (#74172)
openshift/openshift-docs: modules/machine-pools-hcp.adoc

// Module included in the following assemblies:
//
// * rosa_cluster_admin/rosa_nodes/rosa-nodes-machinepools-about.adoc

:_mod-docs-content-type: CONCEPT
[id="machine-pools-hcp_{context}"]
= Machine pools in {hcp-title} clusters

> In {hcp-title} clusters, the {hcp} spans three availability zones (AZ) in the installed cloud region. Each machine pool in a {hcp-title} cluster deploys in a single subnet within a single AZ. Each of these AZs can only have one machine pool. 

Reviewer: the hcp variable resolves to a plural "hosted control planes":
```suggestion
In {hcp-title} clusters, the {hcp} span three availability zones (AZ) in the installed cloud region. Each machine pool in a {hcp-title} cluster deploys in a single subnet within a single AZ. Each of these AZs can only have one machine pool. 
```

> In {hcp-title} clusters, the {hcp} spans three availability zones (AZ) in the installed cloud region. Each machine pool in a {hcp-title} cluster deploys in a single subnet within a single AZ. Each of these AZs can only have one machine pool. 

Reviewer: Per the [ISG](https://www.ibm.com/docs/en/ibm-style?topic=grammar-adverbs-only) we should move "only" as close to the phrase it modifies if possible. If this sentence means that AZs are limited to one machine pool, I'd change it to this:
```suggestion
In {hcp-title} clusters, the {hcp} spans three availability zones (AZ) in the installed cloud region. Each machine pool in a {hcp-title} cluster deploys in a single subnet within a single AZ. Each of these AZs can have only one machine pool. 
```

> In {hcp-title} clusters, the {hcp} spans across three availability zones (AZ) in the installed cloud region. Each machine pool in a {hcp-title} cluster deploys in a single subnet within a single AZ. Each of these AZs can only have one machine pool. 

Reviewer 2: ```suggestion
In {hcp-title} clusters, the {hcp} spans three availability zones (AZ) in the installed cloud region. Each machine pool in a {hcp-title} cluster deploys in a single subnet within a single AZ. Each of these AZs can only have one machine pool. 
```

Each machine pool in an {hcp-title} cluster upgrades independently. Because the machine pools upgrade independently, they must remain within 2 minor (Y-stream) versions of the hosted control plane. For example, if your hosted control plane is 4.16.z, your machine pools must be at least 4.14.z.

> The following images depict how machine pools work within ROSA and {hcp-title} clusters:

Reviewer: Nit - is this technically just one image? Or are you referring to the fact that this image file has 3 separate sections of diagrams?
```suggestion
The following image depicts how machine pools work within ROSA and {hcp-title} clusters:
```

Ignore this if your wording was intentional

> Additionally, machine pools in {hcp-title} clusters each upgrade independently and the versions of the machine pools must remain within two minor (Y-stream) versions of the control plane. While machine pools created with your cluster use the same version, since machine pools upgrade independently, the machine pools must remain within 2 y-stream versions of the hsted control plane.

Reviewer 2: ```suggestion
Each machine pool in an {hcp-title} cluster upgrades independently. Because the machine pools upgrade independently, they must remain within 2 minor (Y-stream) versions of the hosted control plane.
```
It might behoove you to add a more practical example as well, such as: "For example, if your hosted control plane is 4.16.z, your machine pools must be at least 4.14.z" (or something along those lines).

image::hcp-rosa-machine-pools.png[Machine pools on ROSA classic and ROSA with HCP clusters]

[NOTE]
====

> Unlike ROSA classic machine pools, which match the control plane's Y-stream versions, machine pools in {hcp-title} clusters each upgrade independently and the versions of the machine pools must remain within two minor (Y-stream) versions of the control plane.

Reviewer 2: ```suggestion
Machine pools in {hcp-title} clusters each upgrade independently and the machine pool versions must remain within two minor (Y-stream) versions of the control plane.
```
I'm not sure if the comparison to ROSA Classic is necessary. If it a customer switching from Classic, it could be useful but otherwise, it just bogs down the text.

// Module included in the following assemblies:
//
// * rosa_cluster_admin/rosa_nodes/rosa-nodes-machinepools-about.adoc

:_mod-docs-content-type: CONCEPT
[id="machine-pools-hcp_{context}"]
= Machine pools in {hcp-title} clusters

> In {hcp-title} clusters, the {hcp} spans all three of the availability zones (AZ) in the installed cloud region. Each machine pool in a {hcp-title} cluster deploys in a single subnet within a single AZ. Additionally, machine pools in {hcp-title} clusters each upgrade independently and the versions of the machine pools must remain within two minor (Y-stream) versions of the control plane. 

Reviewer 3: all three... suggest mentioning as 
spans across three Availability Zones. 

Because some regions have > 3 AZs.

Reviewer 3: _Each machine pool in a {hcp-title} cluster deploys in a single subnet within a single AZ._

Each machine pool in a {hcp-title} cluster deploys in a single subnet within a single AZ. And, each AZ can only have one machine pool.

Reviewer 3: _Additionally, machine pools in {hcp-title} clusters each upgrade independently and the versions of the machine pools must remain within two minor (Y-stream) versions of the control plane._

Here, we're not talking about upgrades of machine pool. That'll be covered in the Cluster Upgrade section. We should only mention about machine pool version here. 

When you create a machine pool, it will be at the same OpenShift x.y.z version as the HCP. However, when the HCP is upgraded, there is not need to upgrade the machine pools. Thus, over time, machine pool can remain within 2 Y-versions from HCP's version.
